apps/emqx_ft/src/emqx_ft_schema.erl

```diff
@@ -93,36 +93,30 @@ fields(file_transfer) ->
             )},
         {storage,
             mk(
-                hoconsc:union(
-                    fun
-                        (all_union_members) ->
-                            [ref(local_storage)];
-                        ({value, #{<<"type">> := <<"local">>}}) ->
-                            [ref(local_storage)];
-                        ({value, #{<<"type">> := _}}) ->
-                            throw(#{field_name => type, expected => "local"});
-                        ({value, _}) ->
-                            [ref(local_storage)]
-                    end
-                ),
+                ref(storage_backend),
                 #{
+                    desc => ?DESC("storage_backend"),
                     required => false,
-                    desc => ?DESC("storage"),
-                    default => #{<<"type">> => <<"local">>}
+                    validator => validator(backend),
+                    default => #{
+                        <<"local">> => #{}
+                    }
                 }
             )}
     ];
-fields(local_storage) ->
+fields(storage_backend) ->
     [
-        {type,
+        {local,
```

Comment:
do we plan to add another storage backend in the future ?

Reply:
We discussed a storage that uploads files to S3 on the fly.
We decided not to implement it in the first iterations because it would be quite complicated algorithmically